src/sync/app.hpp

```diff
@@ -280,12 +260,23 @@ class App {
     std::string m_app_route;
     std::string m_auth_route;
     uint64_t m_request_timeout_ms;
+
+    void refresh_access_token(std::function<void(Optional<AppError>)> completion_block) const;
+
+    void handle_auth_failure(const AppError& error,
```

Comment:
Some doc for this method would be nice. It seems to do a lot of things that are not immediate obvious from the function name?

Reply:
docs added